src/index.js

```diff
@@ -18,57 +18,70 @@ const createDir = (path) => {
   }
 };
 
+const getFreePort = async () => {
+  try {
+    return await getPort();
+  } catch (error) {
+    logger.error('Failed to get a free port', error);
+    return null;
+  }
+};
+
 export default class Differencify {
   constructor(conf) {
     this.configuration = sanitiseGlobalConfiguration(conf);
     this.chromeInstances = {};
-    this.chromeInstancesId = CHROME_PORT;
     if (this.configuration.debug === true) {
       logger.enable();
     }
     createDir(this.configuration.screenshots);
     createDir(this.configuration.testReportPath);
   }
 
-  _createChromeInstance(testConfig) {
+  async _createChromeInstance(testConfig) {
     const width = testConfig.resolution.width || CHROME_WIDTH;
     const height = testConfig.resolution.height || CHROME_HEIGHT;
     const flags = [`--window-size=${width},${height}`];
+    const port = await getFreePort();
+    if (!port) {
+      return null;
+    }
     const chromy = new Chromy({
+      port,
       chromeFlags: flags,
-      port: this.chromeInstancesId,
       waitTimeout: this.configuration.timeout,
       visible: this.configuration.visible,
     });
     return chromy;
   }
 
-  _updateChromeInstances(id, chromy) {
-    this.chromeInstances[id] = chromy;
-    this.chromeInstancesId += 1;
+  _updateChromeInstances(chromy) {
+    this.chromeInstances[chromy.options.port] = chromy;
   }
 
-  async _closeChrome(id, chromy) {
+  async _closeChrome(chromy, testName) {
     try {
-      logger.log('closing browser');
+      logger.prefix(testName).log('closing browser');
       await chromy.close();
-      delete this.chromeInstances[id];
+      delete this.chromeInstances[chromy.options.port];
     } catch (error) {
-      logger.error(error);
+      logger.prefix(testName).log(error);
     }
   }
 
   async _run(config, type, step) {
     const testConfig = sanitiseTestConfiguration(config);
-    const chromy = this._createChromeInstance(testConfig);
-    const testId = this.chromeInstancesId;
-    this._updateChromeInstances(testId, chromy);
+    const chromy = await this._createChromeInstance(testConfig);
+    if (!chromy) {
+      return false;
+    }
+    this._updateChromeInstances(chromy);
     testConfig.type = type;
     if (step) {
```

Comment:
I know this isn't part of this commit but should step just be pushed into the test config by the `test` method before calling `_run`? It would be cleaner.